### PR TITLE
Add unit tests for CryptoKey

### PR DESCRIPTION
**Pull request by wheatear-dev**
Further to #43440, this Pull Request adds several unit tests for CryptoKey, as a second attempt after the related #88988 .

> CryptoKey Load and save a .key file.

These concern a `CryptoKey` instance that has loaded in the attached keys, under `tests/data/crypto/`:

- `load()` a private/public key is respected by `is_public_only()`
- default `save()` leads to identical file contents for `tests/data/crypto/in.key` and `tests/data/crypto/in.pub`, since they are the same key
- `save()` with `public_only` set to `true` leads to identical contents as the separately prepared public key, namely `tests/data/crypto/in.pub`

Additionally, `DOCTEST_CONFIG_NO_EXCEPTIONS_BUT_WITH_ALL_ASSERTS` was added to the `CPPDEFINES` of the `mbedtls` module, if both tests are enabled **and** exceptions are disabled. This was so that the `REQUIRE` assert could be used.

### CODE REVIEW

File: modules/mbedtls/tests/test_crypto_mbedtls.cpp [hunk 1]
```diff
@@ -60,4 +61,42 @@ void hmac_context_digest_test(HashingContext::HashType ht, String expected_hex)
 	String hex = String::hex_encode_buffer(digest.ptr(), digest.size());
 	CHECK(hex == expected_hex);
 }
+
+Ref<CryptoKey> create_crypto_key(String key_path, bool public_only) {
```
Comment (AThousandShips) [inline modules/mbedtls/tests/test_crypto_mbedtls.cpp:65]:
```suggestion
Ref<CryptoKey> create_crypto_key(const String &p_key_path, bool p_public_only) {
```
For all

Comment (wheatear-dev) [reply modules/mbedtls/tests/test_crypto_mbedtls.cpp]: Hi @AThousandShips 👋 

I have actioned this. Please double check though! 😄

**Review by AThousandShips — APPROVED** [review]
Looks good to me

CC @Faless

**Review by Faless — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**